compiler/luci/pass/src/SubstituteExpandDimsToReshapePass.test.cpp

```diff
@@ -247,6 +253,26 @@ TEST_F(SubstituteExpandDimsToReshapeTest, simple_with_expand_dims_2)
   ASSERT_EQ(1, reshape_shape->at<loco::DataType::S32>(3));
 }
 
+TEST_F(SubstituteExpandDimsToReshapeTest, scalar_axis)
+{
+  _graph.init({16, 3, 1}, {16, 3, 1, 1}, 2);
+
+  _graph.scalar_axis(2);
```

Comment:
IDK how it works, that,

This part is called in `init` and
```cpp
    _axis = luci::create_const_node<int32_t>(g(), loco::DataType::S32, {1}, axis);
    _axis->name("axis");
```

This part is called again,
```cpp
    _axis = luci::create_const_node<int32_t>(g(), loco::DataType::S32, {}, axis);
    _axis->name("axis");
```

Is the later node with scalar axis is overriding the previous dim1 node?

Reply:
Yes. It creates a new scalar node and replaces `_axis` with it.